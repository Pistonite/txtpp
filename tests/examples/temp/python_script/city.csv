City,Country
Tokyo,Japan
Delhi,India
Shanghai,China
Sao Paulo,Brazil
Mumbai,India
Mexico City,Mexico
Beijing,China
Osaka,Japan
Cairo,Egypt
New York,United States