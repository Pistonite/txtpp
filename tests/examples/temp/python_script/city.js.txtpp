// In this example we will export a python script,
// and use that to generate JS code from a csv
function get_cities() {
    return [
        // TXTPP#temp gen_cities.g.py
        // import csv
        //
        // with open('city.csv', 'r') as f:
        //     reader = csv.reader(f)
        //     next(reader) # skip header
        //     for city, country in reader:
        //         print(f"{{ city: \"{city}\", country: \"{country}\"}},")
        /* --- generated code --- */
        // TXTPP#run python gen_cities.g.py
        /* --- generated code --- */
    ];
}
// Note we used /*  */ to break the prefix pattern `// ` so that the directive can end
// You can also use an empty line to make it simple.
