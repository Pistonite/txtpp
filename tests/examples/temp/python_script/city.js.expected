// In this example we will export a python script,
// and use that to generate JS code from a csv
function get_cities() {
    return [
        /* --- generated code --- */
        { city: "Tokyo", country: "Japan"},
        { city: "Delhi", country: "India"},
        { city: "Shanghai", country: "China"},
        { city: "Sao Paulo", country: "Brazil"},
        { city: "Mumbai", country: "India"},
        { city: "Mexico City", country: "Mexico"},
        { city: "Beijing", country: "China"},
        { city: "Osaka", country: "Japan"},
        { city: "Cairo", country: "Egypt"},
        { city: "New York", country: "United States"},
        /* --- generated code --- */
    ];
}
// Note we used /*  */ to break the prefix pattern `// ` so that the directive can end
// You can also use an empty line to make it simple.
